docs/modules/ROOT/pages/guides/v4-migration/authorization.adoc

```diff
@@ -0,0 +1,272 @@
+});
+----
+
+With the new implementation, the library expects the `Authorization` header to be extracted in the format of a bearer token in the `token` field of the context:
+
+[source, typescript, indent=0]
+----
+const server = new ApolloServer({
+    schema, // schema from Neo4jGraphQL.getSchema()
+});
+
+const { url } = await startStandaloneServer(server, {
+    listen: { port: 4000 },
+    context: async ({ req }) => ({
+        token: req.headers.Authorization,
+    }),
+});
+----
+
+This is to acknowledge the fact that there are a variety of servers which don't have a `req` object (such as serverless functions, which use `event`). This is a uniform experience, no matter what server you use.
+
+=== `rolesPath`
+
+`rolesPath` was used to configure a custom path for the "roles" claim in the JWT structure. This configuration has now been moved into the type definitions themselves. So given a previous instantiation:
+
+[source, typescript, indent=0]
+----
+new Neo4jGraphQL({
+    typeDefs,
+    plugins: {
+        auth: new Neo4jGraphQLAuthJWTPlugin({
+            secret: "secret",
+            rolesPath: "some.nested.path",
+        }),
+    }
+})
+----
+
+This will now need to instead be configured in the type definitions as:
+
+[source, graphql, indent=0]
+----
+type JWT @jwt {
+    roles: [String!]! @jwtClaim(path: "some.nested.path")
+}
+----
+
+The type name itself can be anything, as long as it is decorated by `@jwt`.
+
+Whilst this is more boilerplate, the strongly typed nature of the definition means there is significantly more powerful filtering options in version 4.0.0.
+
+== Global authentication
+
+Global authentication was previously configured in the auth plugin consructor, for instance:
+
+[source, typescript, indent=0]
+----
+new Neo4jGraphQL({
+    typeDefs,
+    plugins: {
+        auth: new Neo4jGraphQLAuthJWTPlugin({
+            secret: "secret",
+            globalAuthentication: true,
+        }),
+    }
+})
+----
+
+To remain consistent with the use of directives for configuration, this is now achieved in type definitions by extending the schema:
+
+[source, graphql, indent=0]
+----
+extend schema @authentication
+----
+
+== Rules
+
+=== `allow`
+
+Given an `allow` rule, which checks the `id` field of a `User` against the JWT subject _before_ any operation:
+
+[source, graphql, indent=0]
+----
+type User @auth(rules: [{ allow: { id: "$jwt.sub" } }]) {
+    id: ID!
+}
+----
+
+This will now become:
+
+[source, graphql, indent=0]
+----
+type User @authorization(validate: [{ when: [BEFORE], where: { node: { id: "$jwt.sub" } } }]) {
+    id: ID!
+}
+----
+
+Note that `allow` is no longer a discrete rule, but configured by a `when` argument which is an array accepting values `BEFORE` and `AFTER`.
+
+It is our view that users will quite rarely need to specify this argument as it defaults to both, and users will often want to validate a node property both before and after each operation.
```

Comment:
We shouldn't use first person here, but this one is hard to be adapted since it's a disclosure of the team's decision. Maybe:

```suggestion
It is expected that users rarely need to specify this argument as it defaults to both.
Instead, users may prefer to validate a node property both before and after each operation.
```

Reply:
I made a slight tweak, with slightly different wording more similar to the original intention!